common/rpc/encryption/localStoreTlsFactory.go

```diff
@@ -64,15 +81,39 @@ func NewLocalStoreTlsProvider(tlsConfig *config.RootTLS) (TLSConfigProvider, err
 		workerProvider = internodeWorkerProvider
 	}
 
-	return &localStoreTlsProvider{
+	provider := &localStoreTlsProvider{
 		internodeCertProvider:              internodeProvider,
 		internodeClientCertProvider:        internodeProvider,
 		frontendCertProvider:               &localStoreCertProvider{tlsSettings: &tlsConfig.Frontend},
 		workerCertProvider:                 workerProvider,
 		frontendPerHostCertProviderFactory: newLocalStorePerHostCertProviderFactory(tlsConfig.Frontend.PerHostOverrides),
 		RWMutex:                            sync.RWMutex{},
 		settings:                           tlsConfig,
-	}, nil
+		scope:                              scope,
+	}
+	provider.init()
+	return provider, nil
+}
+
+func (s *localStoreTlsProvider) init() {
```

Comment:
I would avoid using `init` as a func name. It is build-in go func to init package.